### DISCUSSION TITLE
Error compiling iPXE - fatal error: config/local/console.h: No such file or directory

### DISCUSSION THREAD

**Discussion by uptivuptiz** [ipxe/ipxe#382] [Q&A] [answered]
Hi,
I've been trying to compile the source of iPXE with WSL running Ubuntu.
As I am somewhat a novice at Linux and this is the first program ever I've tried to compiled, I ask that you bear with me.

What I've done:
- Followed the guide on the [iPXE website](https://ipxe.org/download).
- Installed GCC, binutils, make, perl, liblzma or xz header files, mtools, genisoimage and syslinux.
- Found a post about a similar issue on the [depricated forum](https://forum.ipxe.org/showthread.php?tid=6878).
- Tried making a blank `console.h` file. That did resovle the first error, it just pops up with `umalloc.h` missing next. And `nap.h` after that.

Everything should be up to date as I created this system today.
Console log in this [pastebin](https://pastebin.com/178rqB1a).

I hope this can be solved :)

**Comment by NiKiZe** [comment, marked as answer]
Make sure the time is correct, reclone the repo and try again. If you still have issues, it usually helps if you post more than one line of the output.

But we have seen this in some cases and we are not sure why the config files are not created, but a reclone almost always fixes it. One other thing you could do is to remove the config dir using rm -r config, then md config and git checkout .
Which should reset the time and hopefully force make to run the intended rules.

**Reply by uptivuptiz** [reply]
That seemed to fix it. Thanks for your help.

**Comment by rthawkcom** [comment]
Confirmed.  Something to do with the time settings.  
Fixed the problem by running "timedatectl set-timezone America/Chicago"
NOTE: Also requires "lib-lzma-dev", "mkisofs" and "isolinux" to be installed as well.
Run "git reset --hard" before trying "make" again.

` 
In file included from include/ipxe/nap.h:13,
                 from core/console.c:4:
./config/nap.h:17:10: fatal error: config/local/nap.h: No such file or directory
   17 | #include <config/local/nap.h>
      |          ^~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile.housekeeping:988: bin/console.o] Error 1
`

**Reply by NiKiZe** [reply]
As explained above, you might see the empty config files not being created on rare occasions after git checkout (which is done as part of clones) this is not a timezone issue. (Theory from me is the checked out files getting different timestamps, which confuses make)

The dependencies you are mentioning is listed in the docs. (isolinux and mkisofs only required if building ISOs, but alternatives is ok as well)
